## Final Project Submission

Please fill out:
* Student name: Jeremiah Waiguru
* Student pace: full time
* Scheduled project review date/time: 
* Instructor name: Mwikali
* Blog post URL: N/A


# Overview

This project analyzes movies datasets from various movie rating sites (through Exploratory Data Analysis), to assist the head of Microsoft's new movie studio  in deciding what type of films to create under their new venture.

# Business understanding

Microsoft faces the challenge of competing against industry giants armed with historical industry insights.
To thrive in this competitive industry , microsoft needs to strategically determine the types of movies that are most likely to succeed in terms of profitability,ratings, and optimal release timing.

Business understanding in this scenario involves microsoft recognizing the opportunity to enter the film industry, specifically by establishing a new movie studio .This decision is driven by observation that many successful companies are creating original video content, and microsoft wants to leverage its resources and brand to tap into this market



# Business Objectives

1. Find out if venturing into the firm industry is a viable business idea for microsoft
2. Find out what kind of genres Microsoft needs to create to maximize on profitability
3. Know current market trends and plan how to be market leaders in the industry

# Data Understanding

In this project we will be analyzing data  from these datasets: Box office Mojo, IMDB(Internet Movie Database) and the numbers'.
We'll now load the datasets below;

# Load libraries


In [2]:
# importing packages that we will use in this project
import csv
import pandas as pd
# setting pandas display to avoid scientific notation in the dataframes
pd.options.display.float_format = '{:.2f}'.format
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import sqlite3
# ignore warnings
import warnings
warnings.filterwarnings('ignore')

# 1. Work on the bom.movie_gross.csv dataset

In [7]:
# loading the dataset
movie_gross = pd.read_csv('zippedData/bom.movie_gross.csv.gz')
movie_gross

title      studio  domestic_gross  \
0                                     Toy Story 3          BV    415000000.00   
1                      Alice in Wonderland (2010)          BV    334200000.00   
2     Harry Potter and the Deathly Hallows Part 1          WB    296000000.00   
3                                       Inception          WB    292600000.00   
4                             Shrek Forever After        P/DW    238700000.00   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.         6200.00   
3383                  Edward II (2018 re-release)          FM         4800.00   
3384                                     El Pacto        Sony         2500.00   
3385                                     The Swan  Synergetic         2400.00   
3386                            An Actor Prepares       Grav.         1700.00   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

# Breakdown to understand our data

The DataFrame has 3387 rows and 5 columns.
## Columns:
1. title - the title of the movie
2. studio - where the movie was produced
3. domestic_gross - the domestic gross revenue of the movie in dollars
4. foreign_gross - the foreign gross revenue of the movie in dollars
5. year - the year in which the movie was produced

In [8]:
movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


### from the above information

1. we see the columns have different data types; object, float and integer
2. studio column, domestic_gross and foreign_gross  has missing data
3. foreign_gross has object as the data type yet it should be in float data type

N/B: missing data and the data type problem will be dealt with while cleaning the data

### summary statistics of the domestic gross

In [9]:
# descriptive statistics 
movie_gross['domestic_gross'].describe()

count        3359.00
mean     28745845.07
std      66982498.24
min           100.00
25%        120000.00
50%       1400000.00
75%      27900000.00
max     936700000.00
Name: domestic_gross, dtype: float64

1. The mean of the  columm is approximately 28.75 million dollars
2. The standard deviation is approximately 66.98 million dollars
3. Minimum value in the column is 1oo dollars, this shows that some movies in the dataset made very little money
4. Maximum value in the column is 936.7 million dollars to show that some movies in the dataset made huge profits domestically


In [10]:
movie_gross['foreign_gross'].describe()

count        2037
unique       1204
top       1200000
freq           23
Name: foreign_gross, dtype: object

1. The count is 2037 non_null values  meaning there are missing values  since we expect our count to be 3387.
2. The unique count shows that there are unique values in the column adding upto 1204
3. Top value in the column is 1.2 million dollars and it appears 23 times meaning there are 23 movies that made 1.2 million dollars in the foreign markets
4. freq shows the number of times a top value appears in column.

# 2. Work on the tmdb.movies dataset

In [6]:
# loading the dataset
tmdb_movie = pd.read_csv("C:\\Users\\JEREMY\\Downloads\\MORINGA 1\\dsc-phase-1-project-v2-4\\zippedData\\tmdb.movies.csv.gz", index_col=0)
tmdb_movie

genre_ids      id original_language  \
0          [12, 14, 10751]   12444                en   
1      [14, 12, 16, 10751]   10191                en   
2            [12, 28, 878]   10138                en   
3          [16, 35, 10751]     862                en   
4            [28, 878, 12]   27205                en   
...                    ...     ...               ...   
26512             [27, 18]  488143                en   
26513             [18, 53]  485975                en   
26514         [14, 28, 12]  381231                en   
26515      [10751, 12, 28]  366854                en   
26516             [53, 27]  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1       33.53   2010-11-19   
1                          How to Train Your Dragon       28.73   2010-03-26   
2                                        Iron Man 2       28.52   2010-05-07   
3                                         Toy Story       28.00   1995-11-22   
4                                         Inception       27.92   2010-07-16   
...                                             ...         ...          ...   
26512                         Laboratory Conditions        0.60   2018-10-13   
26513                               _EXHIBIT_84xxx_        0.60   2018-05-01   
26514                                  The Last One        0.60   2018-10-01   
26515                                  Trailer Made        0.60   2018-06-22   
26516                                    The Church        0.60   2018-10-05   

                                              title  vote_average  vote_count  
0      Harry Potter and the Deathly Hallows: Part 1          7.70       10788  
1                          How to Train Your Dragon          7.70        7610  
2                                        Iron Man 2          6.80       12368  
3                                         Toy Story          7.90       10174  
4                                         Inception          8.30       22186  
...                                             ...           ...         ...  
26512                         Laboratory Conditions          0.00           1  
26513                               _EXHIBIT_84xxx_          0.00           1  
26514                                  The Last One          0.00           1  
26515                                  Trailer Made          0.00           1  
26516                                    The Church          0.00           1  

[26517 rows x 9 columns]

In [5]:
tmdb_movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


from the above information;
Our dataset is clean, no missing  data